kafka/README.md

```diff
@@ -144,17 +144,66 @@ az role assignment create --role "Key Vault Crypto User" --assignee <alias>@micr
+
+ However, the old instructions for how to run this demo on Kubernetes using Strimzi are included below. See the following: 
+
+**Create Strimzi Kafka Cluster (Deprecated):** 
 
 Install the Kafka cluster in the Kafka namespace using the following command or following the instructions [here](https://strimzi.io/quickstarts/)
 
 ```bash 
-kubectl create namespace kafka
 kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
 # Apply the `Kafka` Cluster CR file
 kubectl apply -f https://strimzi.io/examples/latest/kafka/kafka-persistent-single.yaml -n kafka 
 ```
 
+**Create Azure Event Hub Resource:**
+
+```bash 
+export EVENT_HUB_NAMESPACE=kafka-demo-ehubns
+export EVENT_HUB_NAME=kafka-demo-topic
+az eventhubs namespace create --name $EVENT_HUB_NAMESPACE --resource-group $RESOURCE_GROUP --location $LOCATION --enable-kafka true --enable-auto-inflate false
+az eventhubs eventhub create --name $EVENT_HUB_NAME --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE --partition-count 10
+```
+
+#### Generate Keys ####
+This step prepares environment variables for running setup-key.sh to generate the keys used for encrypting kafka messgages. 
+These environment variables are also necessary for generating the consumer pod yaml file if you are using Azure Event Hub. 
+
+```bash
+$ export MAA_ENDPOINT="sharedeus.eus.attest.azure.net"
+# <akv/mHSM url> should have the following format
+# <akv-name>.vault.azure.net or <mHSM-name>.managedhsm.azure.net
+$ export AZURE_AKV_RESOURCE_ENDPOINT= <akv-name>.vault.azure.net or <mHSM-name>.managedhsm.azure.net
+```
+
+#### Generate Consumer Pod YAML Files (Only If Using Azure Event Hub Resource)
+
+```bash 
+$ result=$(az eventhubs namespace authorization-rule list --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE)
+$ export EVENT_HUB_AUTH_RULE_NAME=$(echo "$result" | jq -r '.[0].name')
+$ result=$(az eventhubs namespace authorization-rule keys list --resource-group $RESOURCE_GROUP --namespace-name $EVENT_HUB_NAMESPACE --name $EVENT_HUB_AUTH_RULE_NAME)
+$ export EVENTHUB_CONNECTION_STRING="$(echo "$result" | jq -r '.primaryConnectionString')"
+$ export EVENTHUB_BROKER=$EVENT_HUB_NAMESPACE.servicebus.windows.net:9093
+$ export SIDECAR_IMAGE="mcr.microsoft.com/aci/skr:2.7"
+$ export CONSUMER_IMAGE="mcr.microsoft.com/acc/samples/kafka/consumer:2.0"
+$ export PRODUCER_IMAGE="mcr.microsoft.com/acc/samples/kafka/producer:2.0"
+```
+
+Determine whether `envsubst` is already installed. Issue `which envsubst` command. If the result is empty, install `envsubst` using the following command:
+
+```bash
+curl -L https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-`uname -s`-`uname -m` -o envsubst --fail-with-body 
```

Comment:
Can't we just do this with sed or something instead of downloading and installing software that isn't even managed by a package manager?

Reply:
I had suggested a python script for this previously

Reply:
Using sed is kind of complicated because both the image and the pubkey contains special characters such as : and /. For image names, it is easy to deal with but I just experimented with using sed to replace the pubkey and I'm having so much issues. I would suggest sticking with envsubst. I will look at it again.

Reply:
Would `awk` work? 
```bash
export PUBKEY="<key_placeholder>"
awk -v var="PUBKEY" -v val="$PUBKEY" '{
    gsub("\\$"var, val);
    print;
}' producer/producer-ehub.yaml > producer-example.yaml
```

Reply:
thanks. Got this figured out.